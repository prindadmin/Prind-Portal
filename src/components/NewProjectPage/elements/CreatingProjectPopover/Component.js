import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
