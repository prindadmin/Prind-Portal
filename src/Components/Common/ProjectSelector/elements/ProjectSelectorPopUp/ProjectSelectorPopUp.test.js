import React from 'react';
import { shallow } from 'enzyme';


import Component from './ProjectSelectorPopUp'

it('Should render', () => {
  const props = {
    projects: {
      accessibleProjects: {
        projectCreator: [
          {
            data:"1610470800",
            projectId:"CdmProject2021-01-12",
            projectName:"CDM Project",
            projectDescription: null,
            dateTime:"0000000000",
            projectType:"CDM2015Project",
          },
          {
            data:"1610386904",
            projectId:"TestFromForm3Dhsf2021-01-11",
            projectName:"Test from form 3 DHSF",
            projectDescription:"This is me testing if a DHSF project can be created",
            dateTime:"0000000000",
            projectType:"DHSFProject",
          }
        ],
        projectRole: []
      },
      chosenProject: {
        projectId: '1234',
        projectType: "CDM2015Project"
      }
    },
    onCancelPopup: function() {},
    getAccessibleProjects: function() {},
    updateChosenProject: function() {},
    location: {
      pathname: "/occupation/TestFromForm3Dhsf2021-01-11"
    },
    history: {
      push: function() {},
    },
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


// TODO: Currently has spinner; needs to respond to project request (getAccessibleProjects)
// TODO: Give this function a project list to render
it('Should render with two projects from creator', () => {
  const fetchProjectListFunc = jest.fn((projectId, resolve, reject) => {
    console.log(projectId)
    console.log(resolve)
    console.log(reject)
    resolve()
  });

  /*
  const fetchProjectListFunc = function(projectId, resolve, reject) {
    resolve()
  }
  */

  const props = {
    projects: {
      accessibleProjects: {
        projectCreator: [
          {
            data:"1610470800",
            projectId:"CdmProject2021-01-12",
            projectName:"CDM Project",
            projectDescription: null,
            dateTime:"0000000000",
            projectType:"CDM2015Project",
          },
          {
            data:"1610386904",
            projectId:"TestFromForm3Dhsf2021-01-11",
            projectName:"Test from form 3 DHSF",
            projectDescription:"This is me testing if a DHSF project can be created",
            dateTime:"0000000000",
            projectType:"DHSFProject",
          }
        ],
        projectRole: []
      },
      chosenProject: {
        projectId: '1234',
        projectType: "CDM2015Project"
      }
    },
    onCancelPopup: function() {},
    getAccessibleProjects: fetchProjectListFunc,
    updateChosenProject: function() {},
    location: {
      pathname: "/occupation/TestFromForm3Dhsf2021-01-11"
    },
    history: {
      push: function() {},
    },
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});
