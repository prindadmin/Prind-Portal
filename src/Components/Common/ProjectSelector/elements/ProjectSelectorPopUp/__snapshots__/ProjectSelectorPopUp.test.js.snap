// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<div
  id="popup-greyer"
  onClick={[Function]}
>
  <div
    id="popup-box"
    onClick={[Function]}
  >
    <div
      className="projects-pop-up-header"
    >
      <h2>
        Select a project
      </h2>
    </div>
    <div
      className="project-scroll-box"
    >
      <div
        className="projects-loading-container fill"
      >
        <div
          className="loading-spinner"
        >
          <Blueprint3.Spinner
            intent="danger"
            size={100}
          />
          <p>
            Loading projects, please wait...
          </p>
        </div>
      </div>
    </div>
    <div
      className="projects-pop-up-footer"
    >
      <Blueprint3.Button
        active={true}
        icon="cube-add"
        intent="success"
        large={true}
        onClick={[Function]}
        text="Create a new project"
      />
    </div>
  </div>
</div>
`;

exports[`Should render with two projects from creator 1`] = `
<div
  id="popup-greyer"
  onClick={[Function]}
>
  <div
    id="popup-box"
    onClick={[Function]}
  >
    <div
      className="projects-pop-up-header"
    >
      <h2>
        Select a project
      </h2>
    </div>
    <div
      className="project-scroll-box"
    >
      <div
        className="projects-loading-container fill"
      >
        <div
          className="loading-spinner"
        >
          <Blueprint3.Spinner
            intent="danger"
            size={100}
          />
          <p>
            Loading projects, please wait...
          </p>
        </div>
      </div>
    </div>
    <div
      className="projects-pop-up-footer"
    >
      <Blueprint3.Button
        active={true}
        icon="cube-add"
        intent="success"
        large={true}
        onClick={[Function]}
        text="Create a new project"
      />
    </div>
  </div>
</div>
`;
